File: packages/style-value-parser/lib/properties/transform.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.Transform = void 0;
var _tokenParser = require("../token-parser");
var _transformFunction = require("../css-types/transform-function");
class Transform {
  constructor(value) {
    this.value = value;
  }
  toString() {
    return this.value.join(' ');
  }
  static get parse() {
    return _tokenParser.TokenParser.oneOrMore(_transformFunction.TransformFunction.parser).separatedBy(_tokenParser.TokenParser.tokens.Whitespace).map(value => new Transform(value));
  }
}
exports.Transform = Transform;

File: crates/stylex-css-parser/src/properties/transform.rs
/*!
CSS Transform property parsing.

Handles transform property syntax with multiple transform functions.
*/

use crate::{css_types::transform_function::TransformFunction, token_parser::TokenParser};
use std::fmt::{self, Display};

/// CSS transform property value
#[derive(Debug, Clone, PartialEq)]
pub struct Transform {
  pub value: Vec<TransformFunction>,
}

impl Transform {
  /// Create a new Transform
  pub fn new(value: Vec<TransformFunction>) -> Self {
    Self { value }
  }

  pub fn parser() -> TokenParser<Transform> {
    use crate::token_parser::tokens;

    TokenParser::one_or_more_separated_by(TransformFunction::parse(), tokens::whitespace())
      .map(Transform::new, Some("to_transform"))
  }
}

#[cfg(not(tarpaulin_include))]
impl Display for Transform {
  fn fmt(&self, f: &mut fmt::Formatter<'_>) -> fmt::Result {
    let function_strings: Vec<String> = self.value.iter().map(|func| func.to_string()).collect();
    write!(f, "{}", function_strings.join(" "))
  }
}

#[cfg(test)]
#[path = "../tests/properties/transform_tests.rs"]
mod tests;

#[cfg(test)]
#[path = "../tests/properties/transform_test.rs"]
mod transform_test;
